src/main/java/at/gammastrahlung/monopoly_server/game/PaymentSystem.java

```diff
@@ -0,0 +1,131 @@
+package at.gammastrahlung.monopoly_server.game;
+
+import at.gammastrahlung.monopoly_server.game.gameboard.*;
+
+
+// Class to handle all payment transactions within the game
+public class PaymentSystem {
+
+    private final GameBoard gameBoard; // GameBoard instance for accessing game fields
+
+    // Constructor to initialize PaymentSystem with a GameBoard
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class PaymentSystem {
+
+    private final GameBoard gameBoard;
+
+
+    public PaymentSystem(GameBoard gameBoard) {
+        this.gameBoard = gameBoard;
+    }
+
+
+    // Method to process payments for railroads
+    public boolean processRailroadPayment(Player payer, Railroad railroad) {
+        if (railroad.getOwner() != null && !railroad.getOwner().equals(payer)) {
+            int ownedRailroads = countOwnedRailroads(railroad.getOwner());
+            String key = ownedRailroads + "RR";
+            if (railroad.getRentPrices().containsKey(key)) {
+                int rentAmount = railroad.getRentPrices().get(key);
+                return makePayment(payer, railroad.getOwner(), rentAmount);
+            }
+
+    private boolean processRailroadPayment(Player payer, Railroad railroad) {
+        if (!railroad.getOwner().equals(payer)) {
+            int ownedRailroads = countOwnedRailroads(railroad.getOwner());
+            // Zugriff auf die RentPrices über den von Lombok generierten Getter
+            int rentAmount = railroad.getRentPrices().get(ownedRailroads + "RR");
+            return makePayment(payer, railroad.getOwner(), rentAmount);
+
+        }
+        return false;
+    }
+
+
+    // Method to process payments for properties
+    public boolean processPropertyPayment(Player payer, Property property) {
+        if (property.getOwner() != null && !property.getOwner().equals(payer)) {
+            int rentAmount = property.getRentPrices().getOrDefault("0", 0); // Using string key "0" if using a map with string keys
+
+
+    private boolean processPropertyPayment(Player payer, Property property) {
+        if (!property.getOwner().equals(payer)) {
+            int rentAmount = property.getRentPrices().getOrDefault(0, 0); // Default rent if no houses
+
+            return makePayment(payer, property.getOwner(), rentAmount);
+        }
+        return false;
+    }
+
+
+    // Method to process payments for utilities
+    public boolean processUtilityPayment(Player payer, Utility utility) {
+        if (utility.getOwner() != null && !utility.getOwner().equals(payer)) {
+            int rentAmount = utility.getToPay();  // Assumes getToPay() gives the correct amount due
+
+
+
+    private boolean processUtilityPayment(Player payer, Utility utility) {
```

Comment:
Same method signature as the public method, this will not compile,
same for some other methods, I will not add further comments.